content/docs/features/dockerfiles.md

```diff
@@ -0,0 +1,315 @@
+  care.
+* When Dockerfiles are used to switch the run image from that defined on the provided builder, the resulting run image
+  may not have all the mixins required by buildpacks that detected. Platforms may wish to optionally re-validate mixins
+  prior to `build` when using extensions.
+
+### Phased approach
+
+Some limitations of the initial implementation of the Dockerfiles feature have already been mentioned, and we'll expand
+on them here. As this is a large and complicated feature, the implementation has been split into phases in order to
+deliver incremental value and gather feedback.
+
+* Phase 1 (supported in lifecycle `0.15.0` or greater): one or more `run.Dockerfile`s each containing a single `FROM`
+  instruction can be used to switch the original run image to a new image (as no image modifications are permitted,
+  there is no need to run `extend` on the run image)
+* Phase 2 (supported in lifecycle `0.15.0` or greater): one or more `build.Dockerfile`s can be used to extend the
+  builder image
+  * A new `extend` lifecycle phase is introduced to apply `build.Dockerfile`s from `generate` to the builder image
+* Phase 3 (future): one or more `run.Dockerfile`s can be used to extend the run image
+  * The `extend` lifecycle phase can be run in parallel for the builder and run images
+
+The final ordering of lifecycle phases will look something like the following:
+
+* `analyze`
+* `detect` - after standard detection, `detect` will also run extensions' `./bin/generate`; output Dockerfiles are
+  written to a volume
+* `restore`
+* `extend` - applies one or more `build.Dockerfile`s to the builder image
+* `extend` - applies one or more `run.Dockerfile`s to the run image (could run in parallel with builder image extension)
+* `build`
+* `export`
+
+The [pack cli](https://github.com/buildpacks/pack)
+and [Tekton task](https://github.com/tektoncd/catalog/tree/main/task/buildpacks-phases/0.2) have or will eventually have
+support for builds using image extensions. However, [kpack](https://github.com/pivotal/kpack) - while able to use image
+extensions in builds - will need additional updating to propagate changes such as a switched run image back to the
+appropriate resource to avoid breaking rebase.
+
+## In action: a CNB build with extensions
+
+Let's walk through a build that uses extensions, step by step. We will see an image extension that installs `curl` on
+the builder image, and switches the run image to an image that has `curl` installed.
+
+### 0. Setup workspace directory
+
+* `workspace=<your preferred workspace directory>`
+
+### 1. Clone the pack repo and build it (TODO: remove when pack with extensions-phase-2 support is released)
+
+* `cd $workspace`
+* `git clone git@github.com:buildpacks/pack.git`
+* `cd pack`
+* `git checkout extensions-phase-2`
+* `make clean build`
+
+### 2. Clone the samples repo
+
+* `cd $workspace`
+* `git clone git@github.com:buildpacks/samples.git`
```

Comment:
```suggestion
* `git clone https://github.com/buildpacks/samples.git`
```

We probably want to use the "https" links since the git protocol ones are private.